src/main/java/com/minecolonies/coremod/blocks/AbstractBlockDoor.java

```diff
@@ -0,0 +1,63 @@
+package com.minecolonies.coremod.blocks;
+
+import com.minecolonies.api.util.constant.Suppression;
+import com.minecolonies.coremod.blocks.interfaces.IBlockMinecolonies;
+import com.minecolonies.coremod.items.ModItems;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockDoor;
+import net.minecraft.block.material.Material;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemBlock;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.world.World;
+import net.minecraftforge.registries.IForgeRegistry;
+
+import java.util.Random;
+
+public class AbstractBlockDoor<B extends AbstractBlockDoor<B>> extends BlockDoor implements IBlockMinecolonies<B>
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this class abstract or rename it, since it matches the regular expression '^Abstract[A-Z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00118)